### PR TITLE
Procgen asteroids for salvage magnet

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

Uses moony's ~~incomprehensible~~ very epic system to add procgen asteroids for salvage.
The asteroids have a 60% chance to spawn in place of a conventional salvage map.

Sloth wanted the asteroids fuckin huge so here you go.
massive.

Additionally cleans up salvagesystem (not all the way)
Adjusts the salvage spawning algorithm to actually offset it from the magnet
reduces the range to compensate for this offset.
Adjusts the salvage algorithm to pick the map/asteroid first *then* pick the location.

**Media**

image
image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: The salvage magnet can now pull in randomly generated asteroids stuffed with ores.
- fix: Salvages are much less likely to fail to be pulled in.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Would be nice if these were a lot bigger and less dense.

**Comment by EmoGarbage404** [comment]
> Would be nice if these were a lot bigger and less dense.

image

i have it generating larger ones.

having them be less dense is fine but if there's no ore they're kinda uninteresting. I'll try increasing standard rock up to like 40% and see how that looks.

**Comment by EmoGarbage404** [comment]
image
image
image
image
@metalgearsloth bigger shapes, less dense

**Comment by metalgearsloth** [comment]
Should be at least like 30x30 with hollow bits but maybe I'm alone in feeling that they're tiny af.

File: Content.Server/Salvage/SalvageSystem.cs [hunk 1]
```diff
@@ -296,125 +296,125 @@ private void SafeDeleteSalvage(EntityUid salvage)
...
+                return false;
+            }
 
-            for (var i = 0; i < allSalvageMaps.Count; i++)
+            if (salvageEnt is { } ent)
+            {
+                var salvXForm = Transform(ent);
+                _transform.SetParent(ent, salvXForm, _mapManager.GetMapEntityId(spawnLocation.MapId));
+                _transform.SetWorldPosition(salvXForm, spawnLocation.Position);
+                RaiseLocalEvent(ent, new LocalStructureLoadedEvent());
```
Comment (moonheart08) [inline Content.Server/Salvage/SalvageSystem.cs:370]: sus amogus
just make LocalityLoaderSystem always load the object if there is no world controller.

**Review by moonheart08 — CHANGES_REQUESTED** [review]

**Review by moonheart08 — APPROVED** [review]

**Pull request merged**